homeassistant/util/yaml.py

```diff
@@ -119,10 +124,49 @@ def _env_var_yaml(loader, node):
         raise HomeAssistantError(node.value)
 
 
+# pylint: disable=protected-access
+def _secret_yaml(loader, node):
+    """Load secrets and embed it into the configuration YAML."""
+    # Create secret cache on loader and load secret.yaml
+    if not hasattr(loader, '_SECRET_CACHE'):
+        loader._SECRET_CACHE = {}
+
+    secret_path = os.path.join(os.path.dirname(loader.name), 'secrets.yaml')
+    if secret_path not in loader._SECRET_CACHE:
+        if os.path.isfile(secret_path):
+            loader._SECRET_CACHE[secret_path] = load_yaml(secret_path)
+            secrets = loader._SECRET_CACHE[secret_path]
+            if 'logger' in secrets:
+                logger = str(secrets['logger']).lower()
+                if logger == 'debug':
+                    _LOGGER.setLevel(logging.DEBUG)
+                else:
+                    _LOGGER.error("secrets.yaml: 'logger: debug' expected,"
+                                  " but 'logger: %s' found", logger)
+                del secrets['logger']
+        else:
+            loader._SECRET_CACHE[secret_path] = None
+    secrets = loader._SECRET_CACHE[secret_path]
+
+    # Retrieve secret, first from secrets.yaml, then from keyring
+    if secrets is not None and node.value in secrets:
+        _LOGGER.debug('Secret %s retrieved from secrets.yaml.', node.value)
+        return secrets[node.value]
+    elif keyring:
+        # do ome keyring stuff
+        pwd = keyring.get_password(_SECRET_NAMESPACE, node.value)
+        if pwd:
+            _LOGGER.debug('Secret %s retrieved from keyring.', node.value)
+            return pwd
+
+    _LOGGER.error('Secret %s not defined.', node.value)
```

Comment:
I think that here it would be helpful if message provide some more details because the issue can be at various places (not `secrets.yaml` file, no keyring, no access to the keyring, or no entry).